### PR TITLE
feat: onboarding

### PR DESCRIPTION
**Pull request by DefK1lla**

### CODE REVIEW

File: src/components/DataSourceForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const { Title, Text } = Typography;
+
+const DataSourceForm: FC = () => {
+  const [activeTile, setActiveTile] = useState<string>();
+  const [keyword, setKeyword] = useState<string>("");
+
+  return (
+    <div className={styles.wrapper}>
+      <Title level={3}>New Data Source</Title>
```
Comment (ifokeev) [inline src/components/DataSourceForm/index.tsx:20]: let's use https://react.i18next.com/latest/usetranslation-hook for all strings, we'll need to connect multiple languages

**Review by ifokeev — CHANGES_REQUESTED** [review]

File: src/components/DataSourceForm/data.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import Postgres from "../../assets/databases/postgre.svg";
```
Comment (ifokeev) [inline src/components/DataSourceForm/data.tsx:1]: could we use `@` symbol here?

```
import Postgres from "@/assets/databases/postgre.svg";
```

File: src/components/DataSourceForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import DataSourceSelection from "../DataSourceSelection";
```
Comment (ifokeev) [inline src/components/DataSourceForm/index.tsx:1]: could we use `@` symbol here?

File: src/components/DataSourceForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import { dbTiles } from "./data";
+
+import type { DataSourceOption } from "../DataSourceSelection";
```
Comment (ifokeev) [inline src/components/DataSourceForm/index.tsx:5]: could we use `@` symbol here?

**Review by ifokeev — CHANGES_REQUESTED** [review]
let's use `@` symbol for every relative path

File: src/components/DataSourceForm/index.tsx (hunk 3 above)
Comment (DefK1lla) [reply src/components/DataSourceForm/index.tsx:1]: Only @/components/DataSourceSelection

Need to do?

Comment (ifokeev) [reply src/components/DataSourceForm/index.tsx:1]: yes please

File: src/components/DataSourceSetup/index.module.less [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+.submit {
+  width: 141px;
+
+  margin-right: var(--gap-5xs);
+
+  @media (max-width: 576px) {
+    width: 100%;
+    margin-right: 0;
+    margin-bottom: var(--gap-5xs);
+  }
```
Comment (ifokeev) [inline src/components/DataSourceSetup/index.module.less:77]: such media queries should be solved via antd grid system, it's possible. Using media queries is allowed in very rare cases when it's not possible to use the grid system

**Review by ifokeev — CHANGES_REQUESTED** [review]

**Comment by ifokeev** [comment]
Let's combine `useResponsive` from ahooks https://ahooks.js.org/hooks/use-responsive, Grid from `antd` and `classnames` to create adaptive layouts

File: src/components/ApiSetup/index.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    <div>
+      <Title level={3}>Setup SQL API</Title>
+
+      <Text>
+        You can connect any BI tool or other SQL compatible software using the
+        details specified below
+      </Text>
+
+      <Form layout="vertical">
+        <Form.Item label="Data source" className={styles.label}>
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:61]: could we use https://github.com/react-hook-form/react-hook-form?

File: src/components/ApiSetup/index.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    />
+  );
+};
+
+const textareaValue = `MYSQL  --host=gh-api.clickhouse.tech
+    - -user=user@api.clickhouse.tech
+    - -port=5121
+    - -password=**********`;
+
+const ApiSetup: FC = () => {
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:50]: form state (input values) should be provided via props

File: src/components/DataModelGeneration/index.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+const items = [
+  {
+    key: "1",
+    label: "default",
+    children: (
+      <Row gutter={[8, 8]}>
+        <Col span={24}>
+          <div className={styles.field}>
+            <Checkbox>trips {"→"} Trips.js</Checkbox>
```
Comment (ifokeev) [inline src/components/DataModelGeneration/index.tsx:30]: tables and measures/dimensions should be provided via props

something like object:

```
tableName: {
  measures: {
   ....
  },
  dimensions: { 
  ....
  }
}
```

please check in data schemas on demo account
https://app.mlcraft.org/~/schemas/7d7eea55-ad47-4456-8a1b-badef7cf7270

demo credentials are provided in the mlcraft repo

File: src/components/DataSourceSetup/index.tsx [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+        </Text>
+      </div>
+      <Form
+        className={styles.form}
+        layout="vertical"
+        id="setup-form"
+        onFinish={console.log}
+      >
+        <Form.Item className={styles.label} label="Name*" name="name">
+          <Input placeholder="gh-api.clickhouse.tech (Yandex Demo)" />
```
Comment (ifokeev) [inline src/components/DataSourceSetup/index.tsx:109]: values should be provided via form state from props

**Review by ifokeev — CHANGES_REQUESTED** [review]

File: src/components/DataSourceForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,136 @@
+import { Suspense } from "react";
+
+import DataSourceSelection from "@/components/DataSourceSelection";
+import type { DataSource, DynamicForm } from "@/types/dataSource";
+
+import DataSourceSetup from "../DataSourceSetup";
```
Comment (ifokeev) [inline src/components/DataSourceForm/index.tsx:6]: let's use `@/components/componentName` for all paths, it's easier to debug and change

File: src/components/DataSourceSetup/index.tsx [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+                  defaultValue={defaultValue}
+                  value={value}
+                  onChange={(e) => onChange(e.target.value)}
+                />
+              </Form.Item>
+            )}
+          />
+        );
+    }
+  };
```
Comment (ifokeev) [inline src/components/DataSourceSetup/index.tsx:129]: let's create a component from this

File: src/components/DataSourceForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+      default:
+        return (
+          <DataSourceSelection
+            onSubmit={onDataSourceSelect}
+            initialValue={formData?.dataSource}
+            options={dbTiles}
+          />
+        );
+    }
+  };
```
Comment (ifokeev) [inline src/components/DataSourceForm/index.tsx:163]: let's create a component from this

File: src/components/DataModelGeneration/index.tsx [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+                              ({schema[s][tb].length}){" "}
+                              {t("common:words.columns")}
+                            </span>
+                          </div>
+                        </Form.Item>
+                      )}
+                    />
+                  </div>
+                ))}
+              </Panel>
```
Comment (ifokeev) [inline src/components/DataModelGeneration/index.tsx:124]: is it possible to create a component for this codebase?

File: src/components/ApiSetup/index.tsx [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+        <Controller
+          name="name"
+          control={control}
+          defaultValue={initialValue?.name}
+          render={({ field: { onChange, value } }) => (
+            <Form.Item label="Data source" className={styles.label}>
+              <Input value={value} onChange={(e) => onChange(e.target.value)} />
+            </Form.Item>
+          )}
+        />
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:82]: possible a component called `input`

File: src/components/ApiSetup/index.tsx [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+                    value={o.value}
+                    disabled={o.disabled}
+                  >
+                    {o.label}
+                  </Radio>
+                ))}
+              </Radio.Group>
+            </Form.Item>
+          )}
+        />
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:109]: possible a component called `radioGroup`

File: src/components/ApiSetup/index.tsx [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+                            onClick={() =>
+                              navigator.clipboard.writeText(f.value)
+                            }
+                          />
+                        }
+                      />
+                    )}
+                  </Form.Item>
+                )}
+              />
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:139]: possible a component called `password`

File: src/components/DataSourceFormBody/index.tsx [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+              disabled: false,
+              name: "connection",
+            },
+            {
+              value: "psql",
+              label: "PSQL",
+              disabled: false,
+              name: "connection",
+            },
+          ]}
```
Comment (ifokeev) [inline src/components/DataSourceFormBody/index.tsx:145]: it's default values without providing in the props

File: src/components/ApiSetup/index.tsx [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+  const { control, handleSubmit, getValues, watch, resetField } =
+    useForm<ApiSetupForm>();
+
+  const { t } = useTranslation(["apiSetup", "common"]);
+  const windowSize = useResponsive();
+
+  const createConnectionString = useCallback(
+    (
+      connection: string = CONNECTION_DEFAULT
+    ) => `${connection}  --host=gh-api.clickhouse.tech
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:63]: host is hardcoded, will be provided in the ENV

File: src/components/ApiSetup/index.tsx [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+                    <CopyIcon
+                      className={styles.icon}
+                      onClick={() =>
+                        navigator.clipboard.writeText(getValues(name) || "")
+                      }
+                    />
+                  }
+                />
+              </Col>
+            );
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:129]: could be provided without props

File: src/components/ApiSetup/index.stories.tsx [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+    <ApiSetup {...args} />
+  </RootLayout>
+);
+
+export const Default = Template.bind({});
+
+Default.args = {
+  connectionOptions: [
+    { value: "mysql", label: "MySQL", disabled: false, name: "connection" },
+    { value: "psql", label: "PSQL", disabled: false, name: "connection" },
```
Comment (ifokeev) [inline src/components/ApiSetup/index.stories.tsx:23]: could be hardcoded cuz we have only mysql and postgresql interfaces

File: src/components/DataSourceSetup/index.tsx [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+          <Button
+            className={cn(styles.link, styles.skip)}
+            type="link"
+            onClick={onSkip}
+          >
+            {t("common:words.skip")}
+          </Button>
+        </Row>
+      </Form>
+      <Alert
```
Comment (ifokeev) [inline src/components/DataSourceSetup/index.tsx:133]: should be shown if any error provided, not visible by default for all views

File: src/components/ApiSetup/index.tsx [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+            htmlType="submit"
+            form="api-setup"
+            onClick={handleSubmit(onSubmit)}
+          >
+            {t("common:words.finish")}
+          </Button>
+
+          <Button className={styles.link} type="link" onClick={onDownload}>
+            {t("common:words.download_credentials")}
+          </Button>
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:199]: is it possible to use `space` component here? https://ant.design/components/space

File: src/components/ApiSetup/index.tsx [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+            type="primary"
+            size="large"
+            htmlType="submit"
+            form="api-setup"
+            onClick={handleSubmit(onSubmit)}
+          >
+            {t("common:words.finish")}
+          </Button>
+
+          <Button className={styles.link} type="link" onClick={onDownload}>
```
Comment (ifokeev) [inline src/components/ApiSetup/index.tsx:197]: could you add downloading credentials json file from the browser?

**Review by ifokeev — APPROVED** [review]

**Pull request merged**